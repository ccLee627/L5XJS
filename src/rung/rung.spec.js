const Rung = require("./index");

describe("Rung Class", () => {
    describe("New Instance", () => {
        it("Accepts Proper Input", () => {
            const fn = (logic, num, comment = null) => () => new Rung(logic, num, comment);

            expect(fn("XIC(someTag)NOP();")).toThrow();
            expect(fn("XIC(someTag)NOP();", "notANum")).toThrow();
            expect(fn(null, 12)).toThrow();
            expect(fn("XIC(someTag)NOP();", 12)).not.toThrow();
            expect(fn("XIC(someTag)NOP();", 12, 12)).toThrow();
            expect(fn("XIC(someTag)NOP();", 12, "I am a Comment")).not.toThrow();
        });

        it("Initializes with Desired Document Model", () => {
            let rung = new Rung("XIC(someTag)NOP();", 12);
            expect(rung).toMatchSnapshot();

            rung = new Rung("XIC(someTag)NOP();", 12, "I am a comment");
            expect(rung).toMatchSnapshot();
        });
    });

    describe("Static Methods", () => {
        test("isRung: Returns Appropriate Judgement", () => {
            let rung = new Rung("XIC(someTag)NOP();", 0, "A Comment");
            expect(Rung.isRung(rung)).toBeTruthy();

            rung = { notARung: 12 };
            expect(Rung.isRung(rung)).toBeFalsy();

            rung = 12;
            expect(Rung.isRung(rung)).toBeFalsy();
        });
    });
});
