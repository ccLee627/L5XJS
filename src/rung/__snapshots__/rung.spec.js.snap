// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rung Class New Instance Initializes with Desired Document Model 1`] = `
Rung {
  "_dom": Object {
    "attributes": Object {
      "Number": 12,
      "Type": "N",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "XIC(someTag)NOP();",
            "type": "cdata",
          },
        ],
        "name": "Text",
        "type": "element",
      },
    ],
    "name": "Rung",
    "type": "element",
  },
  "class_id": "b49e5b79e8662ca3b6e884d3057719d0",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Rung Class New Instance Initializes with Desired Document Model 2`] = `
Rung {
  "_dom": Object {
    "attributes": Object {
      "Number": 12,
      "Type": "N",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "I am a comment",
            "type": "cdata",
          },
        ],
        "name": "Comment",
        "type": "element",
      },
      Object {
        "elements": Array [
          Object {
            "cdata": "XIC(someTag)NOP();",
            "type": "cdata",
          },
        ],
        "name": "Text",
        "type": "element",
      },
    ],
    "name": "Rung",
    "type": "element",
  },
  "class_id": "b49e5b79e8662ca3b6e884d3057719d0",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;
